graspologic/plot/plot.py

```diff
@@ -671,6 +676,226 @@ def pairplot(
+    label_palette : dict, optional, default: dictionary using 'Set1'
+    cluster_palette : dict, optional, default: dictionary using 'Set1'
+    alpha : float, optional, default: 0.7
+        Opacity value of plotter markers between 0 and 1
+
+    References
+    ----------
+    .. [1] Weiss, R., Varoquaux, G.(2011). Scikit-learn: Machine Learning in PythonJournal
+        of Machine Learning Research, 12, 2825–2830.
+    """
+    sns.scatterplot(
+        data=data, x=X[:, j], y=X[:, k], ax=ax, hue="labels", palette=label_palette
+    )
+    for i, (mean, covar) in enumerate(zip(means, covariances)):
+        v, w = linalg.eigh(covar)
+        v = 2.0 * np.sqrt(2.0) * np.sqrt(v)
+        u = w[0] / linalg.norm(w[0])
+        # Plot an ellipse to show the Gaussian component
+        angle = np.arctan(u[1] / u[0])
+        angle = 180.0 * angle / np.pi
+        ell = mpl.patches.Ellipse(
+            [mean[j], mean[k]],
+            v[0],
+            v[1],
+            180.0 + angle,
+            color=cluster_palette[i],
+        )
+        ell.set_clip_box(ax.bbox)
+        ell.set_alpha(alpha)
+        ax.add_artist(ell)
+        # removes tick marks from off diagonal graphs
+        ax.set(xticks=[], yticks=[], xlabel=k, ylabel=k)
+        ax.legend().remove()
+
+
+def pairplot_with_gmm(
+    X,
+    gmm,
+    labels=None,
+    cluster_palette=None,
+    label_palette=None,
+    title=None,
+    context="talk",
+    font_scale=1,
+    alpha=0.7,
+    figsize=(12, 12),
+    kde=True,
+):
+    r"""
+    Plot pairwise relationships in a dataset.
+    By default, this function will create a grid of Axes such that each
+    dimension in data will by shared in the y-axis across a single row and
+    in the x-axis across a single column.
+
+    Parameters
+    ----------
+    X : np.ndarray (2D)
+        input graph
+    gmm: Gaussian Mixture Model object
+    labels : 1d np.ndarray or list, same length as dimensions of X
+        Labels for different categories of graph nodes
+        If labels are not passed in then labels assume the same value as
+        the gmm clusters each point as
+    label_palette : str or dict, optional,
+        with no input : dictionary using 'Set1'
+    cluster_palette : str or dict, optional, default: dictionary using 'Set1'
+        with no input : dictionary using 'Set1'
+    title : string, no default
+    context :  None, or one of {talk (default), paper, notebook, poster}
+        Seaborn plotting context
+    font_scale : float, optional, default: 1
+        Separate scaling factor to independently scale the size of the font
+        elements.
+    alpha : float, optional, default: 0.7
+        Opacity value of plotter markers between 0 and 1
+    figsize : tuple
+        the size of the 2d subplots configuration
+    kde : boolean
+        allows for access of KDE argument in histplot
+    """
+    # Handle X and labels
+    if labels is not None:
+        check_X_y(X, labels)
```

Comment:
just curious if these do what we think they do here, can you add a super quick test for passing in the wrong types/shapes of inputs in various ways?

Reply:
@bdpedigo I tested it out in the notebook to check if it errors with bad size matching and ti does.  and within the test_plot.py code i also have a testing input function that tests if not right input.